target/linux/ath79/dts/qca9558_buffalo_wzr-450hp2.dts

```diff
@@ -127,35 +127,187 @@
 &mdio0 {
 	status = "okay";
 
-	phy0: ethernet-phy@0 {
+	switch@0 {
+		compatible = "qca,qca8337";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
 		reg = <0>;
+		qca,ignore-power-on-sel;
+		qca,led-open-drain;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				ethernet = <&eth0>;
+				phy-mode = "rgmii-id";
+				tx-internal-delay-ps = <1000>;
```

Comment:
I wonder if these are needed.

https://github.com/openwrt/openwrt/pull/15926#discussion_r1689299662 Says these are default

Reply:
phy-mode cannot be omitted, otherwise
```
[   14.837825] qca8k mdio.0:00: OF node /ahb/eth@19000000/mdio/switch@0/ports/port@0 of CPU port 0 lacks the required "phy-mode" property
[   14.850238] qca8k: probe of mdio.0:00 failed with error -22
```

On the other hand, delays could be omitted.